Marketplace/AI_Nav/AI_Navigation.js

```diff
@@ -0,0 +1,356 @@
+
+    // Handle the events we're recieving from the web UI
+    function onWebEventReceived(event) {
+        print("AI_Navigation_App.js received web event: " + event);
+        // Converts string event to object
+        info = JSON.parse(event);
+        if ((info.type =="Preview") && ((beaconNum <= 8) && (beaconNum > 1)) && (model == undefined)) {
+            Script.update.connect(MakeMove);
+            //set idle 
+            Character_Model = info.Model_URL;
+            Idle_time = info.Idle_time;
+            Idle_Anim = info.Idle_Animation_URL;
+            Moving_Anim = info.Moving_Animation_URL;
+            speed = info.Speed;
+            //get index of last beacon added to domain
+            var lastBeaconIndex = beacons.length - 1;
+            //Create the model
+            var modelProperties = {
+                "name": "NAV-AnimatedModel",
+                "type": "Model",
+                "position": Entities.getEntityProperties(beacons[lastBeaconIndex]).position,
+                "lifetime": -1,
+                "damping": .9,
+                "collisionless": true, //maybe change to false
+                "userData": "{\"grabbableKey\":{\"grabbable\":false}}",
+                animation: {
+                    running: true,
+                    url: info.Moving_Animation_URL
+                },
+
+                collidesWith: "",
+                modelURL: info.Model_URL,
+                dynamic: true,
+                grabbable: false,
+                shapeType: "compound"
+            };
+            model = Entities.addEntity(modelProperties);
+            //Change the registration point of the model so that the model appears on the beacon instead of in the middle of it.
+            modelDim = Entities.getEntityProperties(model).dimensions;
+            var raiseModel = {
+                "registrationPoint": {
+                    x: .5,
+                    y: (modelDim.y / 2),
+                    z: 0
+                }
+            };
+            Entities.editEntity(model, raiseModel);
+            //Allows model to begin moving
+            move = true;
+        } else if ((info.type =="Preview") && (beaconNum <= 8) && (beaconNum > 1) && (model != undefined)) {
+            //if a model is already created this allows it to get changed instantly
+            Character_Model = info.Model_URL;
+            Idle_time = info.Idle_time;
+            Idle_Anim = info.Idle_Animation_URL;
+            Moving_Anim = info.Moving_Animation_URL;
+            speed = info.Speed;
+            move = true;
+            var modelProperties = {
+                animation: {
+                    url: info.Moving_Animation_URL
+                },
+                modelURL: info.Model_URL
+            };
+            Entities.editEntity(model, modelProperties);
+        } else if ((info.type =="Preview") && ((beaconNum > 8) || (beaconNum < 2))) {
+            Window.alert("Must be 2-8 beacons");    

```

Comment:
This shouldn't be Window.alert - this should appear in the tablet directly, not as part of a separate UI window that uses the desktop UI